lib/sendingDomains.js

```diff
@@ -0,0 +1,175 @@
+ * of things, like you can't specify rfc822 content and then specify html/plaintext
+ *
+ * @param input object Flat object of configuration for the sending domain to be sent
+ * @returns object
+ */
+var toApiFormat = function(input) {
+  var model = {
+    dkim: {}
+  };
+
+  model.domain = input.domain;
+  model.dkim['private'] = input.privateKey;
+  model.dkim['public'] = input.publicKey;
+  model.dkim.selector = input.selector;
+  model.dkim.headers = input.headers;
+
+  return model;
+};
+
+/**
+ * Private Method for putting URL construction logic into one place
+ *
+ * @returns string Fully constructed URL for contacting the Sending Domains API
+ */
+var constructURL = function () {
+  return config.options.protocol + '://' + config.options.host + (config.options.port ? ':' + config.options.port: '') + '/api/' + config.options.version + '/sending-domains';
+};
+
+/**
+ *  Private Method for issuing GET request to Sending Domains API
+ *
+ *  This method is responsible for getting the collection _and_
+ *  a specific entity from the Sending Domains API
+ *
+ *  If DomainName parameter is omitted, then we fetch the collection
+ *
+ *  @param DomainName string Domain Name of specific Sending Domain to retrieve
+ *  @param callback function
+ */
+var fetch = function(domainName, callback) {
+  var cb = callback
+    , options = {
+      url: constructURL(),
+      headers: {
+        authorization: config.options.key
+      },
+      strictSSL: config.options.strictSSL
+    };
+
+  /*
+   * Check that someone passed in a real Domain Name, if not,
+   * then the domainName parameter is acutally our callback
+   */
+  if (typeof domainName !== 'function' && typeof domainName !== undefined) {
+    options.url = options.url + '/' + domainName;
+  } else {
+    cb = domainName;
+  }
+
+  // Fetch, Fido, Fetch! (We may want to abstract the actual HTTP requests in a helper when we have more libs)
+  request.get(options, function(err, res, body) {
+    if (err) {
+      cb(new Error('Unable to contact Sending Domains API: ' + err));
+    } else if (res.statusCode === 404) {
+      cb(new Error('The specified Domain Name does not exist'));
+    } else if (res.statusCode !== 200) {
+      cb(new Error('Received bad response from Sending Domains API: ' + res.statusCode));
+    } else {
+      cb(null, body);
+    }
+  });
+};
+
+/*
+ * "Class" declaration, Sending Domains API exposes five functions, one for creating
+ * a sending domain, another for updating a sending domain, another for verifying
+ * a sending domain, another for getting a list of sending domains, and another for
+ * getting info about a specific sending domain
+ */
+module.exports = {
+  find: function(domainName, callback) { //retrieve
+    fetch(domainName, callback);
+  },
+  all: function(callback) { //list
+    fetch(callback);
+  },
+  create: function(domainBody, callback) {
+    var mappedInput = toApiFormat(domainBody);
```

Comment:
can you create an abstracted method similiar to fetch to handle the post/put.  there is a lot of code here that is duplicated